apiserver/facades/client/spaces/integrity.go

```diff
@@ -0,0 +1,222 @@
+	subnets network.IDSet
+	// spaces is a the target space topology.
+	spaces      network.SpaceInfos
+	appNetworks map[string][]unitNetwork
+	force       bool
+}
+
+// newAffectedNetworks returns a new affectedNetworks reference for
+// verification of the movement of the input subnets to the input space.
+// The input space topology is manipulated to represent the topology that
+// would result from the move.
+func newAffectedNetworks(
+	movingSubnets network.IDSet, spaceName string, currentTopology network.SpaceInfos, force bool,
+) (*affectedNetworks, error) {
+
+	// We need to indicate that any moving fan underlays include
+	// their overlays as being affected by a move.
+	movingOverlays, err := currentTopology.FanOverlaysFor(movingSubnets)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	for _, overlay := range movingOverlays {
+		movingSubnets.Add(overlay.ID)
+	}
+
+	// Now get the topology as would result from moving all of these subnets.
+	newTopology, err := currentTopology.MoveSubnets(movingSubnets, spaceName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &affectedNetworks{
+		subnets:     movingSubnets,
+		spaces:      newTopology,
+		appNetworks: make(map[string][]unitNetwork),
+		force:       force,
+	}, nil
+}
+
+// processMachines iterates over the input machines,
+// looking at the subnets they are connected to.
+// Any machines connected to a moving subnet have their unit networks
+// included for for later verification.
+func (n *affectedNetworks) processMachines(machines []Machine) error {
+	for _, machine := range machines {
+		addresses, err := machine.AllAddresses()
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		var includesMover bool
+		var machineSubnets network.SubnetInfos
+		for _, address := range addresses {
+			// TODO (manadart 2020-04-22): Note that the Subnet method here
+			// looks up the subnet info based on the address CIDR.
+			// The backing schema for this needs to be rethought for multi-net
+			// capability.
+			sub, err := address.Subnet()
+			if err != nil {
+				return errors.Trace(err)
+			}
+			machineSubnets = append(machineSubnets, sub)
+
+			if n.subnets.Contains(sub.ID) {
+				includesMover = true
+			}
+		}
+
+		// We only consider this machine if it has an
+		// address in one of the moving subnets.
+		if includesMover {
+			if err = n.includeMachine(machine, machineSubnets); err != nil {
+				return errors.Trace(err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// includeMachine ensures that the units on the machine and their collection
+// of subnet connectedness are included as affectedNetworks to be validated.
+func (n *affectedNetworks) includeMachine(machine Machine, subnets network.SubnetInfos) error {
+	units, err := machine.Units()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	for _, unit := range units {
+		appName := unit.ApplicationName()
+		unitNets, ok := n.appNetworks[appName]
+		if !ok {
+			n.appNetworks[appName] = []unitNetwork{}
```

Comment:
nit:
```suggestion
			n.appNetworks[appName] = make([]unitNetwork, 0)
```

Reply:
I'll do this in the patch that is on this one's heels.